src/client.ts

```diff
@@ -575,15 +585,30 @@ export class Client {
 		const { limit = Infinity, ...actualParams } = params;
 		const resolvedParams = { pageSize: MAX_PAGE_SIZE, ...actualParams };
 
-		const result = await this.get<TDocument>(resolvedParams);
+		const documents: TDocument[] = [];
+		let latestResult: prismicT.Query<TDocument> | undefined;
+
+		while (
+			(!latestResult || latestResult.page < latestResult.total_pages) &&
+			documents.length < limit
+		) {
+			const start = Date.now();
+			latestResult = await this.get<TDocument>({
+				...resolvedParams,
+				page: latestResult ? latestResult.page + 1 : undefined,
+			});
```

Comment:
Since we're now assuming this process is a long-running one, maybe it's worth getting and providing explicitly the master ref (if none other provided) so that it cannot change during the crawling process? (the 5000ms cache on it might get busted, if another document has been published master ref changes and pagination can be broken)

If so, does the same apply for Integration Field ref (not sure)?

Reply:
Or maybe we can just use the `next_page` value from the previous query?

Reply:
Good idea, should be easy to grab the ref and reuse it for all requests. If someone publishes something during the fetching process, how long is a stale ref valid for?

`next_page` may be more work to use since ideally all queries can still go through the `this.get` method. `next_page` is a URL which would need to be parsed and processed.

Reply:
I didn't make this change yet because I want to make sure using a stale ref is okay. If a `getAll` request takes 10 seconds, is it better to use an updated ref half-way through the loop?

Reply:
I think refs are busted once they are N refs old (N new refs have been created since), we can confirm that tonight ☺️

OK indeed `next_page` is not so straightforward to use!